File: TaskManager/test/src/main/java/org/example/CreateIssueTest.java
package org.example;

import command.CommandManager;
import command.CommandType;
import command.CreateIssueCommand;
import model.Issue;
import model.Priority;
import model.Status;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import repository.IssueRepository;

import static org.junit.jupiter.api.Assertions.*;

public class CreateIssueTest {
    private IssueRepository repository;
    private CommandManager commandManager;
    
    @BeforeEach
    public void setUp() {
        // reset
        repository = IssueRepository.getInstance();
        // clear
        for (String id : repository.getAllIssues().keySet()) {
            repository.removeIssue(id);
        }
        
        commandManager = CommandManager.getInstance();
    }
    
    @Test
    public void CreateIssue_ShouldAddIssueToRepository() {
        // Aadd issue
        Issue issue = new Issue("Test Issue", "This is a test issue", Priority.Medium);
        CreateIssueCommand command = new CreateIssueCommand(issue);
        commandManager.executeCommand(command);
        
        // verify
        assertTrue(repository.containsIssue(issue.getId()));
        Issue retrievedIssue = repository.getIssue(issue.getId());
        assertEquals("Test Issue", retrievedIssue.getTitle());
        assertEquals("This is a test issue", retrievedIssue.getDescription());
        assertEquals(Priority.Medium, retrievedIssue.getPriority());
        assertEquals(Status.Open, retrievedIssue.getStatus()); // Default status
    }
}

File: TaskManager/TaskManager_210042112/TaskManager_112/TaskManagerTests/CreateIssueTest.cs
using TaskManager_112;

namespace TaskManagerTests
{
    public class CreateIssueTest
    {
        [Fact]
        public void CreateIssue_ShouldAddIssueToRepository()
        {
            var repo = new IssueRepository();
            var issue = IssueFactory.CreateIssue(1, "Trying", "Commiting is hard", Priority.Medium);
            var createIssueCommand = new CreateIssueCommand(repo, issue);
            var invoker = new CommandInvoker();

            // Act
            invoker.ExecuteCommand(createIssueCommand);

            // Assert
            var retrievedIssue = repo.GetIssue(1);
            Assert.NotNull(retrievedIssue);
            Assert.Equal("Trying", retrievedIssue.Title);
            Assert.Equal(Priority.Medium, retrievedIssue.Priority);
            Assert.Equal(Status.Open, retrievedIssue.Status);
        }
    }
}